### PR TITLE
chore: upgrades collector to 0.39.0.

### PR DESCRIPTION
**Pull request by jcchavezs**
## Description

This PR upgrades collector to use Otel collector 0.39.0.

### CODE REVIEW

File: exporter/kafkaexporter/kafka_exporter.modified.go [hunk 1]
```diff
@@ -135,7 +135,6 @@ func newSaramaProducer(config Config) (sarama.SyncProducer, error) {
 	c.Metadata.Full = config.Metadata.Full
 	c.Metadata.Retry.Max = config.Metadata.Retry.Max
 	c.Metadata.Retry.Backoff = config.Metadata.Retry.Backoff
-	configureCompression(config.Compression, c)
```
Comment (tim-mwangi) [inline exporter/kafkaexporter/kafka_exporter.modified.go:138]: @laxman-traceable @pavan-traceable Can you verify this is ok?

Comment (pavan-traceable) [reply exporter/kafkaexporter/kafka_exporter.modified.go:138]: @tim-mwangi  I am not the right person, @ravisingal can you please check this ?

Comment (jcchavezs) [reply exporter/kafkaexporter/kafka_exporter.modified.go:138]: cc @laxmanchekka

Comment (laxmanchekka) [reply exporter/kafkaexporter/kafka_exporter.modified.go:138]: Looks good to me.

cc: @jcchavezs @tim-mwangi

**Comment by laxmanchekka** [comment]
@jcchavezs : I remember, in old version, we did two sets of private changes in our fork for kafka configs (compression, batch flush configs). I can see compression related configs are ported here. But I don't see flush related configs.

Just checking we are not missing to port them.

**Comment by jcchavezs** [comment]
@laxmanchekka you mean this one? https://github.com/hypertrace/opentelemetry-collector/pull/48/files it never went in, see https://github.com/hypertrace/opentelemetry-collector/blob/v0.33.0%2Bpatches/exporter/kafkaexporter/config.go cc @japoorv

**Comment by laxmanchekka** [comment]
> @laxmanchekka you mean this one? https://github.com/hypertrace/opentelemetry-collector/pull/48/files it never went in, see https://github.com/hypertrace/opentelemetry-collector/blob/v0.33.0%2Bpatches/exporter/kafkaexporter/config.go cc @japoorv

Oh okay. I was underthe impression this also went in and was looking for relevant merges.
Thanks for clarification. Please proceed with your changes.

**Review by laxmanchekka — APPROVED** [review]

**Comment by tim-mwangi** [comment]
@jcchavezs will you kick off a release for this to trigger an upgrade in saas-dev?

**Pull request merged**